projects/engine/tests/engine.ts

```diff
@@ -202,14 +202,38 @@ describe('Plugin interaction', () => {
     await manager.activatePlugin(['solidity', 'fileManager'])
     await fileManager.call('solidity', 'compile', 'ballot.sol')
     expect(solidity.compile).toHaveBeenCalledWith('ballot.sol')
+    expect(solidity.compile).toHaveBeenCalledTimes(1)
+    expect(solidity['currentRequest']).toBeUndefined()
+  })
+
+  test('Current Request has been updated during call', async () => {
+    await manager.activatePlugin(['solidity', 'fileManager'])
+    let _currentRequest: Plugin['currentRequest']
+    const updateReq = jest.fn((req) => _currentRequest = req)
+    const setCurrentRequest = () => Object.defineProperty(solidity, 'currentRequest', {
+      configurable: true, // getter/setter can be deleted with "delete"
+      get: () => _currentRequest,
+      set: updateReq
+    })
+    // Call with manager
+    setCurrentRequest()
+    await manager.call('solidity', 'compile', 'ballot.sol')
+    expect(solidity['currentRequest']).toBeUndefined()
+    expect(_currentRequest).toEqual({ from: 'manager', path: 'solidity' })
+    // Call with fileManager
+    setCurrentRequest()
+    await fileManager.call('solidity', 'compile', 'ballot.sol')
+    expect(solidity['currentRequest']).toBeUndefined()
+    expect(_currentRequest).toEqual({ from: 'fileManager', path: 'solidity' })
+    expect(updateReq).toHaveBeenCalledTimes(2)
   })
 
   test('Plugin can activate another', async () => {
     await manager.activatePlugin('solidity')
     await solidity.call('manager', 'activatePlugin', 'fileManager')
     const isActive = await manager.isActive('fileManager')
     expect(manager.activatePlugin).toHaveBeenCalledWith('fileManager')
-    expect(manager.canActivate).toHaveBeenCalledWith(solidity.profile, manager.profile)
```

Comment:
It looks like `manager.profile` from`expect(manager.canActivate).toHaveBeenCalledWith(solidity.profile, manager.profile)` was not expected cause the plugin that `solidity` try to activate is the `fileManager` . is this directly part of the fix?

Reply:
could you point me in the PR where the fix is?

Reply:
The test was wrong and it's one reason why it was hard to detect the bug

Reply:
> could you point me in the PR where the fix is?

I don't understand your question

Reply:
if you update the test here (which was before passing I guess), It means that you've added a change in this PR for making sure the test still pass.

Reply:
The change is here : 
https://github.com/ethereum/remix-plugin/pull/171/files/78d3debe2e4c3e2b6bc78e3d4ed3e526fdc865b8#diff-c0e115918fb919bc94b8ec28711c6dd4R17